aFWall/src/main/java/dev/ukanth/ufirewall/util/FingerprintUtil.java

```diff
@@ -0,0 +1,395 @@
+package dev.ukanth.ufirewall.util;
+
+/**
+ * This file was created to simplify Fingerprint APIs and made specifically for (AFWall+) application.
+ * You are free to re-distributed this file anywhere you like. :)
+ * ----------------------------------------------
+ * Created by vzool on 1/20/17.
+ */
+
+import android.Manifest;
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.app.Dialog;
+import android.app.KeyguardManager;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.graphics.Point;
+import android.hardware.fingerprint.FingerprintManager;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.CancellationSignal;
+import android.security.keystore.KeyGenParameterSpec;
+import android.security.keystore.KeyPermanentlyInvalidatedException;
+import android.security.keystore.KeyProperties;
+import android.support.annotation.Nullable;
+import android.support.annotation.RequiresApi;
+import android.support.v4.app.ActivityCompat;
+import android.view.WindowManager;
+import android.widget.TextView;
+
+import java.io.IOException;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+
+import javax.crypto.Cipher;
+import javax.crypto.KeyGenerator;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+
+import dev.ukanth.ufirewall.Api;
+import dev.ukanth.ufirewall.R;
+
+import static android.content.Context.FINGERPRINT_SERVICE;
+import static android.content.Context.KEYGUARD_SERVICE;
+
+public class FingerprintUtil {
+
+    // generate key based on pkg name
+    public static String GetKey(Context context){
+        return Api.getCurrentPackage(context) + ":Fingerprint";
+    }
+
+    // safely check if device support fingerprint
+    public static boolean isAndroidSupport(){
+        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.M;
+    }
+
+    /*
+    * Dialog
+    **/
+    public static class FingerprintDialog extends Dialog{
```

Comment:
It turns out that MaterialDialog does not have a default constructor like what normal dialogs do, it does required a [**Builder**](https://github.com/afollestad/material-dialogs/blob/master/core/src/main/java/com/afollestad/materialdialogs/MaterialDialog.java#L94) not **Context**, which is a complete way of behaviour. I prefer to continue using this dialogs until we discover some way for doing that transition.